src/Nethermind/Nethermind.Facade/Simulate/SimulateDictionaryBlockStore.cs

```diff
@@ -0,0 +1,94 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Collections.Generic;
+using Nethermind.Blockchain.Blocks;
+using Nethermind.Core;
+using Nethermind.Core.Buffers;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Extensions;
+using Nethermind.Serialization.Rlp;
+
+namespace Nethermind.Facade.Simulate;
+
+public class SimulateDictionaryBlockStore(IBlockStore readonlyBaseBlockStore) : IBlockStore
```

Comment:
Minor thing, but technically this has nothing to do with simulate. You could consider moving to Nethermind.Core to make it available for general use.

Reply:
Not while normal HeaderStore is in `Nethermind.Blockchain`=)

Reply:
> Not while normal HeaderStore is in `Nethermind.Blockchain`=)

I was also thinking about naming and namespace.